src/modules/dashboard/staff/components/staff-popup.vue

```diff
@@ -1,5 +1,72 @@
 <template>
-  <div>
-    <div>staff data/form goes here...</div>
-  </div>
+  <StaffForm
+    :action-type="actionType"
+    :staff-id="staffId"
+    @cancel="emit('close')"
+    @confirm="onConfirm"
+  />
 </template>
+
+<script setup lang="ts">
+import { staffService } from '../service/staff.service';
+import { StaffData } from '../models/staff-data';
+import StaffForm from './staff-form.vue';
+import { ActionType } from '../../enums/action-type.enum';
+import { useMutation } from '@tanstack/vue-query';
+import { useGlobalStore } from '@/global/store/global.store';
+import { Message } from '@/global/models/message';
+import { MessageType } from '@/global/enums/message-type.enum';
+import { TranslateResult } from 'vue-i18n';
+import i18n from '@/plugins/i18n';
+
+type StaffPopupEmits = {
+  (e: 'close'): void;
+  (e: 'refetch'): void;
+};
+
+const emit = defineEmits<StaffPopupEmits>();
+
+const props = defineProps({
+  staffId: {
+    type: Number,
+    required: true,
+  },
+  actionType: {
+    type: String as PropType<ActionType>,
+    default: ActionType.ADD,
+  },
+});
+
+const globalStore = useGlobalStore();
+function handleSuccess(messageText: TranslateResult): void {
+  const message = Message.getMessage(MessageType.SUCCESS);
+  message.content = messageText;
```

Comment:
message.content = Message.getMessage(...); nie ma co przelewać z kubka do kubka wartości, jesli nie zabezpieczamy sie tutaj na nieoczekiwane przypadki, nulle czy undefined to nie ma sensu raczej.

Reply:
Chyba nie:D getMessage() zwraca Message, który nie ma contentu. Dopiero przypisuje do niego argument funkcji handleSuccess. Przerabiam to i tak, żeby móc od razu tworzyć message.content w funkcji getMessage() bo w wielu miejscach muszę pisać dodatkowy kod, który za to odpowiada.